govcd/edgegateway.go

```diff
@@ -634,3 +641,97 @@ func (eGW *EdgeGateway) RemoveIpsecVPN() (Task, error) {
 	}
 	return eGW.AddIpsecVPN(ipsecVPNConfig)
 }
+
+// Deletes the edge gateway, returning a task and an error if the vCD call fails.
+// https://code.vmware.com/apis/442/vcloud-director/doc/doc/operations/DELETE-EdgeGateway.html
+func (egw *EdgeGateway) DeleteTask(force bool, recursive bool) (Task, error) {
```

Comment:
Naming: please check in the codebase whether this is the first precedence of providing sync and async function variants. We need to agree on how to call these functions. By this code you're suggesting `x` for sync and `xTask` for async. Personally for me, when I see `DeleteTask` first it makes me think it "deletes some task" :) What about the following scheme:
- `Delete` - synchronous function(s)
- `DeleteAsync` - function(s) that return a task

@Didainius and @vbauzysvmware please chime in with your preferences/thoughts.

Reply:
I am OK with `DeleteAsync`.
I had interpreted your `+1` to Vaidotas request as an endorsement of `DeleteTask`

Reply:
Done

Reply:
I think our intention by default user to choose not to handle tasks - so main function `Delete`.
And another one which returns tasks. `DeleteAsync` - not for my taste, but at the moment don't have better suggestions as ''DeleteAndReturnTask` or others seems worse.
According current govcd, we have `Delete` which returns tasks and `DeleteWait` which handles tasks in some cases.

Reply:
If we voted on hiding task, I was surely against. While I appreciate the simplicity of a function that simply does the job, I want to be able to monitor what is happening. Tasks are an essential part of the API, and I want to use them for monitoring and troubleshooting. 
I can change the name to whatever is decided, but I want to have tasks available for potentially long lasting operations.

Reply:
I agree to leave two functions, no discussion here.

Reply:
After more digging, we already have a precedence for naming *synchronous* tasks:

![Screen Shot 2019-06-11 at 11 17 47](https://user-images.githubusercontent.com/11330689/59255349-9c502280-8c3a-11e9-986e-64a65fe5d9d6.png)

Reply:
Linas, I noted that already :)

Reply:
We decided to have:
1. Default functions (the simply named ones) as synchronous
2. Corresponding asynchronous functions be named as `xAsync`
3. Default synchronous functions returning objects and these should be references (as opposed to copies)

We're not refactoring the whole library now, but use this as a guideline for all future functions.

Reply:
Done!

Functions available for creation:
* `CreateEdgeGateway`: uses a simplified structure, returns an edge gateway and an error
* `CreateAndConfigureEdgeGateway`: uses the full edge gateway structure, returns an edge gateway and an error.
* `CreateEdgeGatewayAsync`: uses a simplified structure, returns a task and an error
* `CreateAndConfigureEdgeGatewayAsync`: uses the full edge gateway structure, returns a task and an error

Delete functions follow the same pattern:
* `EdgeGateway.DeleteAsync` returns a task and an error
* `EdgeGateway.Delete` returns an error